db/gorm.go

```diff
@@ -13,6 +13,29 @@ import (
 // https://mariadb.com/kb/en/identifier-names/#unquoted
 var mariaDBUnquotedIdentifierRE = regexp.MustCompile("^[0-9a-zA-Z$_]+$")
 
+func validMariaDBUnquotedIdentifier(s string) error {
+	if !mariaDBUnquotedIdentifierRE.MatchString(s) {
+		return fmt.Errorf("invalid database column or table name %q", s)
+	}
+
+	allNumeric := true
+	startsNumeric := false
+	for i, c := range []byte(s) {
+		if c < '0' || c > '9' {
+			if startsNumeric && len(s) > i && s[i] == 'e' {
+				return fmt.Errorf("database column or table name looks like floating point notation: %q", s)
```

Comment:
nit here and on similar return on line 34. See also my other comment below:
```suggestion
				return fmt.Errorf("looks like floating point notation: %q", s)
```

Reply:
I recognize that adding error information in the caller rather than the callee is Go style, but in this case I think "looks like floating point notation:" is too vague an error to stand on its own. If a caller ever neglected adding the "table name:" or "column name:" prefix, this error would be nonsensical.

Reply:
Then maybe just "identifier looks like floating point", since that's what this function name says it's checking?